Dockerfile

```diff
@@ -0,0 +1,32 @@
+# ---- Base Alpine with installed Node ----
+FROM alpine:3.14.3 AS base
+
+# install node
+RUN apk add --update nodejs npm nghttp2
+
+# ---- Install dependencies ----
+FROM base AS build
+
+WORKDIR /app
+COPY . .
+# install dependencies
+RUN npm ci
+
+# build to a production Javascript
+RUN npm run build:prod
+
+# ---- Serve ----
+FROM base AS release
+
+WORKDIR /app
+COPY --from=build /app/dist ./dist
+# A wildcard is used to ensure both package.json AND package-lock.json are copied
+# where available (npm@5+)
+COPY package* ./
+# copy OpenaAPI document
+COPY openapi.yaml ./
+# install only production dependencies (defined in "dependencies")
+RUN npm ci --only=production
```

Comment:
In order to use docker cache layer for dependencies, you should run the installation of them isolated after copying the `package*` files. 
It should run before the copy of `openapi.yaml` because Docker will not use the cache layer if that file changed (even though `package*` files didn't).
```suggestion
WORKDIR /app
COPY --from=build /app/dist ./dist
# A wildcard is used to ensure both package.json AND package-lock.json are copied
# where available (npm@5+)
COPY package* ./
# install only production dependencies (defined in "dependencies")
RUN npm ci --only=production
# copy OpenaAPI document
COPY openapi.yaml ./
```